### PR TITLE
feat: add admin mutation

### PR DESCRIPTION
**Pull request by chenjanet**
## Notion ticket link

[Add Admin Mutation](https://www.notion.so/uwblueprintexecs/Add-Admin-Mutation-f95fcd0c4e3b41e2ad2b4e6ac283092a)

## Implementation description
* AddAdminModal.tsx: call the `AddUser` mutation in `onSubmit()` (which runs when the add admin modal form is submitted) to add a new admin user to the database, and added an error alert to the modal that displays when either (a) one or more of the fields is blank upon submitting, or (b) a request error occurs
* AdminConfirmationMessage.tsx, MessageContainer.tsx: updated these two components to be consistent with the Figma designs for the confirmation message styling + button positioning
* typography.ts: added `mobileSubtitle1` as a new text style (as shown [here](https://www.figma.com/file/Uia7C5xwqDlkLnrXMyt5hn/Jump-Math-Design-System?node-id=5%3A2&t=DFdbqB6TzBDnptdU-0)) to be used with the error alerts
* UserTypes.ts: defined new types for the add admin mutation request/response
* UserMutations.ts: created a new `AddUser` mutation to be used by the add admin modal
* Renamed `GetUsersByRole.ts` to `UserQueries.ts` to be consistent with the [folder structure in the Rowan House Society repo](https://github.com/uwblueprint/rowan-house/tree/main/frontend/src/APIClients)
* misc. lint fixes in other files

## Steps to test
1. Click the "+ Add Admin" button on the admin landing page to bring up the add admin modal
2. Fill out the form. Try submitting 
    - with one or more fields blank
    - with an improperly-formatted email
    - with a non-jump math email if the `Yes` radio button is selected
    - with something else that will cause an API request error (you might need to bork the `onSubmit()` code slightly to mock this)
to verify the different failure scenarios as specified in the [Figma design](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=484%3A17091&t=gBJNG7Y2JuEjzspd-0) 
3. Once the form has been submitted successfully (with all inputs valid), connect to MongoDB and verify that a new admin user has been added under the `test.users` collection, with credentials matching those that were inputted

## What should reviewers focus on?
Screen Shot 2022-11-28 at 1 43 28 AM
Screen Shot 2022-11-28 at 1 41 43 AM

[Figma](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=484%3A17091&t=Z3gXSSguA2LMnKB6-0)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/themes/typography.ts [hunk 1]
```diff
@@ -60,6 +60,12 @@ const textStyles = {
     lineHeight: "18px",
     letterSpacing: "-0.02em",
   },
+  mobileSubtitle1: {
```
Comment (joyce-shi) [inline frontend/src/themes/typography.ts:63]: what is `mobileSubtitle1`?

File: frontend/src/types/UserTypes.ts [hunk 2]
```diff
@@ -1,5 +1,23 @@
+import { Role } from "./AuthTypes";
+
 export type AdminUser = {
   firstName: string;
   lastName: string;
   email: string;
 };
+
+export type UserRequest = {
```
Comment (joyce-shi) [inline frontend/src/types/UserTypes.ts]: nit: `UserRequest` -> `NewUserRequest` or `AddUserRequest` or `RegisterUserRequest` (or something to that effect, I don't love any of those names LOL)

File: frontend/src/components/common/AddAdminModal.tsx [hunk 3]
```diff
@@ -15,14 +17,34 @@ import {
...
 import InputError from "./Form/InputError";
 import InputLabel from "./Form/InputLabel";
 import TextInput from "./Form/TextInput";
 import ModalFooterButtons from "./ModalFooterButtons";
 import { PlusOutlineIcon } from "./icons";
+import { ADMIN_PAGE } from "../../constants/Routes";
+import { UserRequest, UserResponse } from "../../types/UserTypes";
+import GET_USERS_BY_ROLE from "../../APIClients/queries/GetUsersByRole";
+
+const ADD_ADMIN = gql`
```
Comment (joyce-shi) [inline frontend/src/components/common/AddAdminModal.tsx]: given that this mutation is called `ADD_ADMIN`, should we set `role = "Admin"` as a default valut? Screenshot attached for code reference on what I mean!

Screen Shot 2022-11-29 at 8 30 51 PM

**Review by joyce-shi — CHANGES_REQUESTED** [review]
Overall LGTM, great work! Thanks for powering through all the design changes (: Left some minor comments.

File: frontend/src/themes/typography.ts (hunk 1 above)
Comment (chenjanet) [reply frontend/src/themes/typography.ts:63]: It's from the [Jump Math Design System](https://www.figma.com/file/Uia7C5xwqDlkLnrXMyt5hn/Jump-Math-Design-System?node-id=12%3A334&t=huzN42tabysdwLnL-0); @vyomeshiyengar said it's the typography used for error states

Comment (joyce-shi) [reply frontend/src/themes/typography.ts:63]: ooo sg! sorry i vyo and i had a miscommunication here.

**Review by joyce-shi — APPROVED** [review]
LGTM!

**Pull request merged**